lib/client.js

```diff
@@ -467,7 +467,14 @@ function Request(method, url) {
       self.emit('response', res);
     }
 
-    self.callback(err, res);
+    if (res && res.status >= 200 && res.status < 300) {
+      return self.callback(err, res);
+    }
+
+    var new_err = new Error('Unsucessful HTTP response');
```

Comment:
I think doing this in node is a no brainer, but on the client I am less sure mostly cause of the giant string dump.